### ISSUE TITLE
[openblas] Build error when using dynamic-arch feature on x64-windows

### ISSUE THREAD

**Issue by edgroc** [microsoft/vcpkg#27575]
**Host Environment**

- OS: Windows 10 Pro 10.0.19044 Build 19044
- Compiler: Visual Studio V16.11.19

**To Reproduce**

./vcpkg install openblas[dynamic-arch]:x64-windows

**Failure logs**

- Cut and paste the appropriate build messages from the console output.

D:\dev\vcpkg>vcpkg.exe install openblas[dynamic-arch]:x64-windows
Computing installation plan...
The following packages will be built and installed:
    openblas[core,dynamic-arch]:x64-windows -> 0.3.21
Detecting compiler hash for triplet x64-windows...
Restored 0 package(s) from C:\Users\Ed\AppData\Local\vcpkg\archives in 336 us. Use --debug to see more details.
Installing 1/1 openblas:x64-windows...
Building openblas[core,dynamic-arch]:x64-windows...
-- Using cached xianyi-OpenBLAS-b89fb708caa5a5a32de8f4306c4ff132e0228e9a.tar.gz.
-- Cleaning sources at D:/dev/vcpkg/buildtrees/openblas/src/32e0228e9a-ba8c0c7bab.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/dev/vcpkg/downloads/xianyi-OpenBLAS-b89fb708caa5a5a32de8f4306c4ff132e0228e9a.tar.gz
-- Applying patch uwp.patch
-- Applying patch fix-space-path.patch
-- Applying patch fix-redefinition-function.patch
-- Applying patch fix-uwp-build.patch
-- Applying patch install-tools.patch
-- Using source at D:/dev/vcpkg/buildtrees/openblas/src/32e0228e9a-ba8c0c7bab.clean
-- Found external ninja('1.10.2').
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:131 (message):
    Command failed: D:/dev/vcpkg/downloads/tools/cmake-3.24.0-windows/cmake-3.24.0-windows-i386/bin/cmake.exe --build . --config Debug --target install -- -v -j9
    Working Directory: D:/dev/vcpkg/buildtrees/openblas/x64-windows-dbg
    See logs for more information:
      D:\dev\vcpkg\buildtrees\openblas\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/openblas/portfile.cmake:71 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)

error: building openblas:x64-windows failed with: BUILD_FAILED
error: Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
Then check for known issues at:
    https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+openblas
You can submit a new issue at:
    https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[openblas]+Build+error
Include '[openblas] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
    vcpkg-tool version: 2022-10-17-3247920fbdd47d08f36cbd480addd9890d3c2435
    vcpkg-scripts version: 05fda317e 2022-10-28 (3 days ago)

**Additional context**

vcpkg install openblas:x64-windows builds without error.
vcpkg install openblas[threads]:x64-windows also builds without error.

**Comment by Adela0814** [comment]
@edgroc  please provide the failure log:D:`\dev\vcpkg\buildtrees\openblas\install-x64-windows-dbg-out.log`, thanks.

**Comment by edgroc** [comment]
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/9906665/install-x64-windows-dbg-out.log)

**Comment by Adela0814** [comment]
Are you using Visual Studio 17.3 Preview 3 or later? Because it only works with it.
see https://github.com/microsoft/vcpkg/pull/26396

**Comment by edgroc** [comment]
I didnt realize or see there was a VS V17.3 requirement.

I ran the vcpkg commands from my x64 Native tools command prompt for VS 2019 (v16.11.19). I assumed, perhaps incorrectly, this would use and build using this compiler version.  

Re-running this with V17.3.6 command promt, I seem to get the same error.
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/9913483/install-x64-windows-dbg-out.log)
 
I also set VCPKG_PLATFORM_TOOLSET to v143 to be sure and re-ran with the same results.
I am attaching the new log file.
